VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,188 @@
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, creationDate),
+                now, answerPrefix + " " + relativeTime + " (label)");
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, creationDate),
+                now, answerPrefix + " " + relativeTime + " by author");
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, null),
+                now, answerPrefix + " " + "by author (label)");
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, creationDate),
+                now, answerPrefix + " " + relativeTime);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, null),
+                now, answerPrefix + " " + "by author");
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, null),
+                now, answerPrefix + " " + "(label)");
+
+        // Empty Case
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, null),
+                now, null);
+    }
+
+    private void assertSetAuthorAttributionText(TextView textView,
+                                                DiscussionTextUtils.AuthorAttributionLabel type,
+                                                IAuthorData input,
+                                                long now, String expectedOutput) {
+        Runnable listener = Mockito.mock(Runnable.class);
+        DiscussionTextUtils.setAuthorAttributionText(textView, type, input, now, listener);
+        if (expectedOutput == null) {
+            assertTrue(textView.getVisibility() == View.GONE);
+        } else {
+            String output = textView.getText().toString();
+            assertEquals(expectedOutput, output);
+            if (!input.isAuthorAnonymous()) {
+                // Test whether author span is clickable or not
+                int start = output.indexOf(input.getAuthor());
+                int end = start + input.getAuthor().length();
+                SpannableString spannableString = (SpannableString) textView.getText();
+                ClickableSpan[] spans = spannableString.getSpans(start, end, ClickableSpan.class);
+                if (config.isUserProfilesEnabled()) {
+                    assertTrue(spans.length == 1);
```

Comment:
By the way, this assertion can be converted into an `assertEquals`, which would provide a better error message upon violation.